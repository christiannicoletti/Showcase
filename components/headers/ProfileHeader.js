import React from "react";
import { View, Image, Text, Platform } from "react-native";
import { useSelector } from "react-redux";
import { HeaderButtons, Item } from "react-navigation-header-buttons";
import HeaderButton from "../UI/IoniconsHeaderButton";
import Menu from "../headers_components/Menu";
import Title from "../headers_components/Title";
import Settings from "../headers_components/Settings";

const TitleOnlyHeader = ({ navigation }) => {
  const darkModeValue = useSelector((state) => state.switches.darkMode);

  return (
    <View>
      <View
        style={{
          padding: 20,
          backgroundColor: darkModeValue ? "black" : "white",
        }}
      />
      <View style={{ flexDirection: "row" }}>
        <Menu
          darkModeValue={darkModeValue}
          navigation={navigation}
          HeaderButton={HeaderButton}
          HeaderButtons={HeaderButtons}
          Item={Item}
          View={View}
          Platform={Platform}
        />
        <Title
          View={View}
          Image={Image}
          Text={Text}
          darkModeValue={darkModeValue}
        />
        <Settings
          darkModeValue={darkModeValue}
          navigation={navigation}
          HeaderButton={HeaderButton}
          HeaderButtons={HeaderButtons}
          Item={Item}
          View={View}
          Platform={Platform}
        />
      </View>
    </View>
  );
};

export default TitleOnlyHeader;
