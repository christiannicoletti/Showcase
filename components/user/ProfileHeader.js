import React from "react";
import { View, Text, StyleSheet, Image, FlatList } from "react-native";
import { useSelector } from "react-redux";
import * as WebBrowser from "expo-web-browser";

import LinkButton from "../UI/LinkButton";
import { TouchableOpacity } from "react-native-gesture-handler";

const Profile = (props) => {
  const userLinks = useSelector((state) => state.projects.userLinks);
  const darkModeValue = useSelector((state) => state.darkMode.darkMode);

  const handleLinkOnPress = (url) => {
    WebBrowser.openBrowserAsync(url);
  };

  return (
    <View
      style={{
        ...styles.container,
        ...props.containerStyle,
      }}
    >
      <View style={{ ...styles.secondContainer }}>
        <View style={{ ...styles.thirdContainer }}>
          <Text style={props.titleStyle}>{props.title}</Text>
          <Text style={props.usernameStyle}>{props.username}</Text>
        </View>
        <Image
          style={{ ...styles.image, ...props.style }}
          source={props.imgSource}
        />
        <TouchableOpacity
          style={{
            margin: 10,
            borderColor: "gray",
            borderWidth: 1,
          }}
          onPress={props.onEditProfilePress}
        >
          <Text
            style={{
              margin: 10,
              color: darkModeValue ? "white" : "black",
            }}
          >
            Edit{"\n"}Profile
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={{
          margin: 10,
          flexDirection: "row",
        }}
      >
        <View
          style={{
            flex: 1,
            borderColor: "gray",
            borderRightWidth: 1,
            borderTopWidth: 1,
            borderBottomWidth: 1,
            alignItems: "center",
          }}
        >
          <Text
            style={{
              margin: 5,
              color: darkModeValue ? "white" : "black",
              fontWeight: "bold",
            }}
          >
            Network
          </Text>
          <Text
            style={{
              marginBottom: 5,
              color: darkModeValue ? "white" : "black",
            }}
          >
            323
          </Text>
        </View>
        <View
          style={{
            flex: 1,
            borderColor: "gray",
            borderBottomWidth: 1,
            borderTopWidth: 1,
            alignItems: "center",
          }}
        >
          <Text
            style={{
              margin: 5,
              color: darkModeValue ? "white" : "black",
              fontWeight: "bold",
            }}
          >
            Advocates
          </Text>
          <Text
            style={{
              marginBottom: 5,
              color: darkModeValue ? "white" : "black",
              fontSize: 15,
            }}
          >
            36
          </Text>
        </View>
      </View>
      <Text style={props.descriptionStyle}>{props.description}</Text>
      <FlatList
        data={userLinks}
        keyExtractor={(item) => item.id}
        numColumns={2}
        renderItem={(itemData) => (
          <LinkButton
            imageUrl={itemData.item.imageUrl}
            title={itemData.item.title}
            textStyle={{ color: darkModeValue ? "white" : "black" }}
            linkContainer={{ borderColor: "gray" }}
            imageStyle={{
              backgroundColor: "white",
              borderRadius: 5,
            }}
            onPress={() => handleLinkOnPress(itemData.item.linkUrl)}
          />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    margin: 5,
  },
  secondContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  thirdContainer: {
    marginRight: 10,
    alignItems: "center",
  },
  image: {
    height: 80,
    width: 80,
    borderRadius: 80 / 2,
  },
});

export default Profile;
