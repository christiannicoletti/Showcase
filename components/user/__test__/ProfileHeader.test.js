import React from "react";
import ProfileHeader from "../ProfileHeader";
import renderer from "react-test-renderer";

import { createStore, combineReducers, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import { SafeAreaProvider } from "react-native-safe-area-context";
import ReduxThunk from "redux-thunk";

import authReducer from "../../../store/reducers/auth";
import switchesReducer from "../../../store/reducers/user";
import userReducer from "../../../store/reducers/user";
import signupReducer from "../../../store/reducers/signup";

test("renders default", () => {
  const rootReducer = combineReducers({
    user: switchesReducer,
    signup: signupReducer,
    auth: authReducer,
    user: userReducer,
  });

  const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

  const tree = renderer
    .create(
      <Provider store={store}>
        <SafeAreaProvider>
          <ProfileHeader />
        </SafeAreaProvider>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders with dummy links", () => {
  const rootReducer = combineReducers({
    user: switchesReducer,
    signup: signupReducer,
    auth: authReducer,
    user: userReducer,
  });

  const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

  const links = {
    ["link1"]: {
      linktitle1: "test",
      linkUrl1: "test",
    },
  };

  const tree = renderer
    .create(
      <Provider store={store}>
        <SafeAreaProvider>
          <ProfileHeader links={links} />
        </SafeAreaProvider>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
