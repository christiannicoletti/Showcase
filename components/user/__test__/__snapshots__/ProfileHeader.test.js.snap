// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`renders with dummy links 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
